src/diffusers/hooks/group_offloading.py

```diff
@@ -68,36 +69,61 @@ def __init__(
         self.buffers = buffers
         self.non_blocking = non_blocking or stream is not None
         self.stream = stream
+        self.record_stream = record_stream
         self.cpu_param_dict = cpu_param_dict
         self.onload_self = onload_self
 
         if self.stream is not None and self.cpu_param_dict is None:
-            raise ValueError("cpu_param_dict must be provided when using stream for data transfer.")
+            raise ValueError("`cpu_param_dict` must be provided when using stream for data transfer.")
+
+        if self.record_stream and not self.stream:
+            raise ValueError("`record_stream` cannot be True when `stream` is None.")
```

Comment:
```suggestion
        if self.stream is not None and self.record_stream:
            raise ValueError("`record_stream` cannot be True when `stream` is None.")
```

(nit): making the check look the same as the one above it for cpu_param_dict